drivers/hmis/spec/models/hmis/form/form_processor_spec.rb

```diff
@@ -1899,13 +1899,146 @@ def add_cde_item_to_definition
+    let(:information_date) { 1.week.ago.to_date }
+    let(:hud_values) do
+      {
+        'CurrentLivingSituation.informationDate' => information_date.strftime('%Y-%m-%d'),
+        'CurrentLivingSituation.currentLivingSituation' => 'SAFE_HAVEN', # 118
+      }
+    end
+
+    it 'should work when CurrentLivingSituation is the form owner' do
+      cls = Hmis::Hud::CurrentLivingSituation.new(client: c1, enrollment: e1, data_source: ds1)
+      process_record(record: cls, hud_values: hud_values, user: hmis_user, definition: definition)
+
+      expect(cls.current_living_situation).to eq(118) # safe haven
+      expect(cls.information_date).to eq(information_date)
+    end
+
+    describe 'when CurrentLivingSituation is being collected on an Assessment,' do
+      let(:definition) { Hmis::Form::Definition.find_by(identifier: 'cls_assessment') }
+      let(:assessment) { build(:hmis_custom_assessment, client: c1, enrollment: e1, data_source: ds1, user: u1, definition: definition, hud_values: hud_values) }
+
+      def process_assessment
+        assessment.form_processor.run!(owner: assessment, user: hmis_user)
+        assessment.form_processor.save!(context: :form_submission)
+      end
+
+      it 'should create a CurrentLivingSituation' do
+        process_assessment
+
+        expect(assessment.current_living_situation).to be_present
+        expect(assessment.current_living_situation.current_living_situation).to eq(118)
+      end
+
+      describe 'and all CurrentLivingSituation fields are hidden:' do
+        before(:each) do
+          assessment.form_processor.hud_values = {
+            'assessmentDate' => Date.current.strftime('%Y-%m-%d'),
+            'CurrentLivingSituation.informationDate' => HIDDEN,
+            'CurrentLivingSituation.currentLivingSituation' => HIDDEN,
+          }
+        end
+
+        it 'should NOT create a CurrentLivingSituation' do
+          expect { process_assessment }.not_to change(e1.current_living_situations, :count)
+          expect(assessment.current_living_situation).to be_nil
+        end
+
+        it 'should destroy the CurrentLivingSituation that was previously attached to the assessment' do
+          # create 2 CurrentLivingSituations for this enrollment
+          cls1 = create(:hmis_current_living_situation, client: c1, enrollment: e1, data_source: ds1, user: u1)
+          create(:hmis_current_living_situation, client: c1, enrollment: e1, data_source: ds1, user: u1)
+
+          # link 1 of the cls to an assessment
+          assessment.save!
+          assessment.form_processor.update(current_living_situation: cls1)
+
+          # check setup
+          expect(assessment.current_living_situation).to eq(cls1)
+          expect(assessment.form_processor.current_living_situation).to eq(cls1)
+
+          # "re-submit" the assessment, this time with all the event fields hidden
+          expect { process_assessment }.to change(e1.current_living_situations.reload, :count).by(-1)
```

Comment:
I don't think you need to reload since relation.count always performs a query afaik (different than relation.size which will use the cache)

Reply:
ah ok thanks